### PR TITLE
feat(e2e): implement aws tests, enhance gcp tests

### LINKED ISSUE #209: e2e tests for AWS Secrets Manager
Implement e2e tests for the AWS SM provider.

AC:

-All happy paths covered
-Few edge cases, and expected failures covered

### PR DESCRIPTION
**Pull request by moolen**
tl;dr
- aws managed tests
- ginkgo v2 migration + labeling
- refactor aws/gcp provider
- removed localstack
- enhance gcp/aws testing: mounted and referenced sa cases
- a lot of small cleanups

Fixes #209 

### Ginkgo v2 labels
With ginkgo v2 we can use labels to select a subset of tests. I labeled the tests with their provider name (orace, gcp, aws, vault...). The "managed" tests have the "managed" label. Our `e2e` test runs all except `managed` tests.  

### Managed test runs

They can be triggered with a comment as per usual. I propose to add a `provider` named parameter. This way we have just one generic job for managed-e2e test runs:

```
/ok-to-test-managed sha=xxxx provider=aws
```

### Cleanups
* removed localstack, we can integrate directly with AWS
* introduced a `newFromEnv()` func for every e2e provider to cleanup the test setup 

### TODO
- [x] setup AWS state bucket `eso-e2e-aws-tfstate` in `eu-west-1` @knelasevero maybe?
- [ ] change back `GCP_KSA_NAME` to non-default `eso-external-secrets` to ensure e2e tests run unprivileged.

### CODE REVIEW

**Comment by moolen** [comment]
I have to merge it to fully test the e2e-managed workflow.

File: .github/workflows/e2e-managed.yml [hunk 1]
```diff
@@ -91,40 +123,52 @@ jobs:
...
+        make tf.show.${PROVIDER}
 
     - name: Setup Infracost
       uses: infracost/actions/setup@v1
       with:
         api-key: ${{ secrets.INFRACOST_API_KEY }}
 
-    - name: Generate Infracost JSON for GKE
-      run: infracost breakdown --path terraform/gcp/plan.json --format json --out-file /tmp/infracost.json
+    - name: Generate Infracost JSON for AWS
```
Comment (knelasevero) [inline .github/workflows/e2e-managed.yml:141]:
```suggestion
    - name: Generate Infracost JSON for provider
```

also using the named arg here, right?

File: e2e/Makefile [hunk 2]
```diff
@@ -21,49 +17,35 @@ start-kind: ## Start kind cluster
 
 test: e2e-image ## Run e2e tests against current kube context
 	$(MAKE) -C ../ docker.build \
-		IMAGE_REGISTRY=local/external-secrets \
```
Comment (knelasevero) [inline e2e/Makefile:24]: Don't we want to use local images for normal e2e tests? If people right now have access to a external secret provider and want to run e2e tests locally they can, without having to push to a registry. Like in this comment: https://github.com/external-secrets/external-secrets/pull/430#issuecomment-975735470

**Comment by knelasevero** [comment]
Let's merge and test from a fork :)

File: e2e/Makefile (hunk 2 above)
Comment (moolen) [reply e2e/Makefile:24]: The `test` target just builds the image locally and sideloads it into the kind cluster. It doesn't get pushed into a registry - it just uses the same name for consistency and ease of use.

Only the `test.managed` target pushes it into the registry.
